sdk/clientcore/core/src/main/java/io/clientcore/core/util/serializer/JsonSerializer.java

```diff
@@ -3,26 +3,65 @@
 
 package io.clientcore.core.util.serializer;
 
+import io.clientcore.core.implementation.TypeUtil;
+import io.clientcore.core.serialization.json.JsonProviders;
+import io.clientcore.core.serialization.json.JsonReader;
+import io.clientcore.core.serialization.json.JsonSerializable;
+import io.clientcore.core.serialization.json.JsonWriter;
+import io.clientcore.core.util.ClientLogger;
+
+import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
+import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Type;
 
 /**
- * Generic interface covering basic JSON serialization and deserialization methods.
+ * Class providing basic JSON serialization and deserialization methods.
```

Comment:
Thoughts on keeping this interface and moving the implementation to DefaultJsonSerializer. 
edit: Offline chat, We may just want to expose ObjectSerializer for the customers to use their own implementations that can include both xml and Json formats.

Reply:
Changed to only expose `ObjectSerializer` for the time being